### PR TITLE
chore: add guild as codeowners of shared files

### PR DESCRIPTION
**Pull request by Kyle-Verhoog**
Now that there are multiple teams with a significant number of devs it makes sense for the guild to start owning some shared files so that package-level changes are signed off by guild members.

## Checklist

- [x] Change(s) are motivated and described in the PR description.
- [x] Testing strategy is described if automated tests are not included in the PR.
- [x] Risk is outlined (performance impact, potential for breakage, maintainability, etc).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] [Library release note guidelines](https://ddtrace.readthedocs.io/en/stable/contributing.html#Release-Note-Guidelines) are followed.
- [x] Documentation is included (in-code, generated user docs, [public corp docs](https://github.com/DataDog/documentation/)).

## Reviewer Checklist

- [x] Title is accurate.
- [x] No unnecessary changes are introduced.
- [x] Description motivates each change.
- [x] Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes unless absolutely necessary.
- [x] Testing strategy adequately addresses listed risk(s).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] Release note makes sense to a user of the library.

### CODE REVIEW

**Review by emmettbutler — APPROVED** [review]

**Comment by mabdinur** [comment]
How is guild membership determined? Will their responsibilities differ from @DataDog/apm-core-python codeowners?

**Review by Yun-Kim — DISMISSED** [review]

File: .github/CODEOWNERS
Comment (brettlangdon) [inline .github/CODEOWNERS:45]: this will overwrite the ASM/core ownership over remote config on L29

they are not additive, it'll take this one instead.

File: .github/CODEOWNERS
Comment (brettlangdon) [inline .github/CODEOWNERS:46]: for this and remoteconfig, should the guild also own the tests?

**Review by brettlangdon — COMMENTED** [review]
should @DataDog/python-guild be a sub-team of `apm-python` ?

should the guild also own the codeowners file?

File: .github/CODEOWNERS
Comment (Kyle-Verhoog) [reply .github/CODEOWNERS:45]: I think that's what we want? The guild should own these.

File: .github/CODEOWNERS
Comment (Kyle-Verhoog) [reply .github/CODEOWNERS:46]: Oh yes, good catch with the tests, that should be the case.

**Comment by Kyle-Verhoog** [comment]
> How is guild membership determined? Will their responsibilities differ from https://github.com/orgs/DataDog/teams/apm-core-python codeowners?

@mabdinur each team has their guild representative(s) (this is already an existing concept) we just never had the need for the guild to explicitly own components of the library. The responsibilities are different from core-python as the guild is responsible for package level things (that affect all/multiple products). The guild is also intended to set the standard for code quality and do architectural review.

**Comment by Kyle-Verhoog** [comment]
> should https://github.com/orgs/DataDog/teams/python-guild be a sub-team of apm-python ?

@brettlangdon Probably? I'm not quite sure what that does though other than some kind of a correlation. Do you know what effect it has?

> should the guild also own the codeowners file?

Yes, good point. I think so.

File: .github/CODEOWNERS
Comment (Kyle-Verhoog) [reply .github/CODEOWNERS:45]: addressed

File: .github/CODEOWNERS
Comment (brettlangdon) [reply .github/CODEOWNERS:44]: I think I agree, just confirming

for any changes to remoteconfig we require approvals from 2 members of the Python guild?


if we add a co-owner like core, then we could do 1 and 1 to widen the pool for approvals.

I'm not saying we have to make what I am suggesting, just talking out loud to confirm expected behavior.

**Comment by brettlangdon** [comment]
> should https://github.com/orgs/DataDog/teams/python-guild be a sub-team of apm-python ?

> @brettlangdon Probably? I'm not quite sure what that does though other than some kind of a correlation. Do you know what effect it has?

@Kyle-Verhoog if someone is a member of the guild, they will also be a member of `apm-python`. so if we have someone from another product team be apart of the guild, they could get notified, and offer review for anything for apm-python.

writing it out, I am not sure that is exactly what we want/require, but... an approach.

**Review by brettlangdon — APPROVED** [review]

File: .github/CODEOWNERS
Comment (Kyle-Verhoog) [reply .github/CODEOWNERS:50]: :+1: let's add core for the less important files so we can have better velocity

**Review by brettlangdon — DISMISSED** [review]
lgtm, we can always adjust later

**Review by emmettbutler — DISMISSED** [review]

**Review by juanjux — APPROVED** [review]

**Review by P403n1x87 — DISMISSED** [review]

**Review by mabdinur — APPROVED** [review]

**Review by Yun-Kim — APPROVED** [review]

**Pull request merged**
